src/main/java/org/spring/FullVersion/CollectionFamilyDao.java

```diff
@@ -38,14 +39,18 @@ public boolean deleteFamily(Family family) {
     @Override
     public void saveFamily(Family family) {
         int index = families.indexOf(family);
-
         if (index >= 0) {
             families.set(index, family);
         } else {
             families.add(family);
         }
 
 
+    }
 
+    @Override
+    public boolean existsById(int index) {
```

Comment:
why we need this method?

Reply:
I have written the calculation logic of the index there